core/txpool/legacypool/list.go

```diff
@@ -281,9 +280,11 @@ type list struct {
 	strict bool       // Whether nonces are strictly continuous or not
 	txs    *sortedMap // Heap indexed sorted hash map of the transactions
 
-	costcap   *big.Int // Price of the highest costing transaction (reset only if exceeds balance)
-	gascap    uint64   // Gas limit of the highest spending transaction (reset only if exceeds block limit)
-	totalcost *big.Int // Total cost of all transactions in the list
+	costCap map[common.Address]*big.Int // Price of the highest costing transaction per currency (reset only if exceeds balance)
+	gascap  uint64                      // Gas limit of the highest spending transaction (reset only if exceeds block limit)
+
+	// Celo additions for multi currency
```

Comment:
`costCap` should be moved here as well if we keep this comment

Reply:
I kept the new map in the same line as the previous costcap intentionally, since it's not an addition but a modification.

Reply:
Fine for me, but then why change `totalcost` to `totalCost`?

Reply:
same reason costcap changed to costCap, to make it more obvious on merge conflicts on the difference between our usage and upstream's.

Reply:
That makes a lot of sense.